sdk/nodejs/runtime/rpc.ts

```diff
@@ -468,6 +480,19 @@ export function deserializeProperty(prop: any): any {
                         [specialSigKey]: specialSecretSig,
                         value: deserializeProperty(prop["value"]),
                     };
+                case specialResourceSig:
+                    const urn = prop["urn"];
+                    // If this value is a URN, create a proxy wrapper around it.
+                    const urnParts = urn.split("::");
+                    const qualifiedType = urnParts[2];
+                    const type = qualifiedType.split("$").pop()!;
+                    const proxyConstructor = proxyConstructors.get(type);
+                    if (proxyConstructor) {
+                        const urnName = urnParts[3];
+                        return new proxyConstructor(urnName, {}, { urn });
+                    }
+                    log.debug(`Saw valid URN ${urn} during deserialization, but no proxy constructor is registered for type ${type}.`);
```

Comment:
seems odd to have this be just a log.  This feels like straight up error that we shoudl throw for.